src/main/java/com/shootoff/camera/cameratypes/PS3EyeCamera.java

```diff
@@ -0,0 +1,621 @@
+
+		return ba;
+
+	}
+
+	public Mat translateCameraArrayToMat(byte[] imageBuffer) {
+		viewHeight = size_y;
+		viewWidth = size_x;
+
+		Mat mat = new Mat(viewHeight, viewWidth, CvType.CV_8UC3);
+
+		mat.put(0, 0, imageBuffer);
+		return mat;
+	}
+
+	public synchronized boolean open() {
+		if (opened)
+			return true;
+		ps3ID = eyecamLib.ps3eye_open(0, size_x, size_y, 75, eyecam.ps3eye_format.PS3EYE_FORMAT_BGR);
+		if (isOpen())
+			closed = false;
+		return isOpen();
+	}
+
+	public Mat getMatFrame() {
+		byte[] frame = getImageNative();
+		currentFrameTimestamp = System.currentTimeMillis();
+		Mat mat = translateCameraArrayToMat(frame);
+		frameCount++;
+		return mat;
+	}
+
+	@Override
+	public BufferedImage getBufferedImage() {
+
+		return Camera.matToBufferedImage(getMatFrame());
+
+	}
+
+	@Override
+	public ShotDetector getPreferredShotDetector(final CameraManager cameraManager, final Configuration config,
+			final CameraView cameraView) {
+		if (OptiTrackShotDetector.isSystemSupported())
+			return new OptiTrackShotDetector(cameraManager, config, cameraView);
+		else if (NativeShotDetector.isSystemSupported())
+			return new NativeShotDetector(cameraManager, config, cameraView);
+		else if (JavaShotDetector.isSystemSupported()) {
+			logger.debug("starting javaShotDetector for PS3Eye");
+			return new JavaShotDetector(cameraManager, config, cameraView);
+		} else
+			return null;
+	}
+
+	public static boolean initialized() {
+		return initialized;
+	}
+
+	static boolean configIsOpen = false;
+
+	@Override
+	public void run() {
+
+		while (isOpen()) {
+			if (cameraEventListener.isPresent())
+				cameraEventListener.get().newFrame(getMatFrame());
+
+			if (((int) (getFrameCount() % Math.min(getFPS(), 5)) == 0) && cameraState != CameraState.CALIBRATING) {
+				estimateCameraFPS();
+			}
+
+			if (configIsOpen) {
+				Platform.runLater(() -> {
+					fpsValue.setText((Double.toString(getFPS()).substring(0, 5)));
+				});
+			}
+
+		}
+		if (cameraEventListener.isPresent())
+			cameraEventListener.get().cameraClosed();
+	}
+
+	private void cameraClosed() {
+		if (cameraEventListener.isPresent())
+			cameraEventListener.get().cameraClosed();
+		close();
+	}
+
+	@Override
+	public boolean isLocked() {
+		return false;
+	}
+
+	private Optional<Integer> origExposure = Optional.empty();
+
+	private int getExposure() {
+		return eyecamLib.ps3eye_get_parameter(ps3ID, eyecam.ps3eye_parameter.PS3EYE_EXPOSURE);
+	}
+
+	private void setExposure(int exposure) {
+		eyecamLib.ps3eye_set_parameter(ps3ID, eyecam.ps3eye_parameter.PS3EYE_EXPOSURE, exposure);
+	}
+
+	@Override
+	public boolean supportsExposureAdjustment() {
+		if (!origExposure.isPresent())
+			origExposure = Optional.of(getExposure());
+		return true;
+	}
+
+	@Override
+	public boolean decreaseExposure() {
+		final int curExp = getExposure();
+		final int newExp = (int) (curExp - (.1 * (double) curExp));
+		logger.debug("curExp[ {} newExp {}", curExp, newExp);
+
+		if (newExp < 20)
```

Comment:
Have you checked to see what the minimal reasonable value for this is for the ps3eye? Unlikely to be the same as the optitrack.  This is the value at which the autocalibration should stop trying to reduce it.  Ultimately it should be in a static variable at the top of the class as well (I will be changing this in optitrack at some point).